R/inject.R

```diff
@@ -0,0 +1,38 @@
+# Evaluates an expression (like `runApp()`) with the shiny.http.response.filter
+# option set to a function which rewrites the <head> to include shiny-tracer.js.
+with_shinytest_js <- function(expr) {
+  shiny::addResourcePath(
+    "shinytest",
+    system.file("js", package = "shinytest")
+  )
+
+  filter <- function(request, response) {
+    if (response$status < 200 || response$status > 300) return(response)
+
+    # Don't break responses that use httpuv's file-based bodies.
+    if ('file' %in% names(response$content))
+      return(response)
+
+    if (!grepl("^text/html\\b", response$content_type, perl=T))
+      return(response)
+
+    # HTML files served from static handler are raw. Convert to char so we
+    # can inject our head content.
+    if (is.raw(response$content))
+      response$content <- rawToChar(response$content)
+
+    # Modify the <head> to load shinytest.js
+    response$content <- sub(
+      "</head>",
+      "<script src=\"shinytest/shiny-tracer.js\"></script>\n</head>",
+      response$content,
+      ignore.case = TRUE
+    )
+
+    return(response)
+  }
+
+  oldOptions <- options(shiny.http.response.filter = filter)
```

Comment:
That's a good idea for chaining the filters. I'll add it even though it's not necessary right now. As far as I know, the only place that `shiny.http.response.filter` is used is in Shiny Server, and if we run tests with Shiny Server and shinytest together, we'll need other changes as well.